app/src/main/java/com/chargebee/example/billing/BillingViewModel.kt

```diff
@@ -26,9 +27,10 @@ class BillingViewModel : ViewModel() {
     var error: MutableLiveData<String?> = MutableLiveData()
     var entitlementsResult: MutableLiveData<String?> = MutableLiveData()
     private var subscriptionId: String = ""
+    private lateinit var sharedPreference : SharedPreferences
 
-    fun purchaseProduct(product: CBProduct, customer: CBCustomer) {
-
+    fun purchaseProduct(context: Context,product: CBProduct, customer: CBCustomer) {
+        sharedPreference =  context.getSharedPreferences("PREFERENCE_NAME",Context.MODE_PRIVATE)
```

Comment:
Can we make this a bit more promiment? 
In essence, this is one of the method in which the customer can choose to cache the product for retrying validation.